### DISCUSSION TITLE
How to setup behind Cloudflare tunnel?

### DISCUSSION THREAD

**Discussion by SlowCar999** [llamafilm/tesla-http-proxy-addon#35] [General]
My ISP provides public IP behind CGNAT, so to access my HA from outside LAN, I have setup Cloudflare tunnel with the HA Cloudflare add-on using my own domain abc.com
I can assess my HA through both ha.abc.com and tesla.abc.com

Following the guide, after filling in domain name and starting tesla proxy addon, log as below:
`s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[19:07:58] INFO: Setting up GnuPG and password-store
gpg: keybox '/data/gnugpg/pubring.kbx' created
gpg: /data/gnugpg/trustdb.gpg: trustdb created
gpg: directory '/data/gnugpg/openpgp-revocs.d' created
gpg: revocation certificate stored as '/data/gnugpg/openpgp-revocs.d/xxxxxxxx.rev'
gpg: checking the trustdb
gpg: marginals needed: 3  completes needed: 1  trust model: pgp
gpg: depth: 0  valid:   1  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 1u
gpg: next trustdb check due at 2027-03-13
/data/gnugpg/pubring.kbx
------------------------
pub   ed25519 2024-03-13 [SC] [expires: 2027-03-13]
      xxxxxxxxx
uid           [ultimate] myself
sub   cv25519 2024-03-13 [E]

created directory: '/data/password-store/'
Password store initialized for myself
[19:08:02] WARNING: Nginx is not running
[19:08:02] INFO: Generating self-signed SSL certificate
-----
[19:08:02] INFO: Generating keypair
-----BEGIN PUBLIC KEY-----
xxxxxxxxxxx
-----END PUBLIC KEY-----
[19:08:02] NOTICE: Request application access with Tesla, then fill in credentials and restart addon.
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped`

Then I start NGINX proxy and get following:

`s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service nginx: starting
s6-rc: info: service nginx successfully started
s6-rc: info: service crond: starting
s6-rc: info: service crond successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[19:13:44] INFO: Generating dhparams (this will take some time)...
Generating DSA parameters, 4096 bit long prime
.....+...+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*
........+...........+...+..+.+......+..................+..........+......+...+................+....+...........................+.+.........+....+......+............+......+.......+...+......+...+.+.....+........+.......+.....+...+.+...+.+..........+....+...........+...+.....+......+..+...............+....+.........+.......xxxxxx.+.......+.+....+...+.........+..............+......+...........+.......+......+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*
[19:13:50] INFO: Running nginx...
stat: can't stat '/ssl/fullchain.pem': No such file or directory
[19:13:50] INFO: Service nginx exited with code 1 (by signal 0)
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service crond: stopping
[19:13:50] INFO: Service crond exited with code 256 (by signal 15)
s6-rc: info: service crond successfully stopped
s6-rc: info: service nginx: stopping
s6-rc: info: service nginx successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped`

Then I got to https://tesla.abc.com/.well-known/appspecific/com.tesla.3p.public-key.pem and get "404: Not Found"

Any idea why?

**Comment by andershagenhansen** [comment]
Installed the Apache2 minimal addon from https://github.com/FaserF/hassio-addons (I'm more familiar with Apache 😊)

Set up Tunnels as this:

![image](https://github.com/llamafilm/tesla-http-proxy-addon/assets/46457331/2877cda3-551e-457b-8c57-7e984345696b)

Set up Apache like this:
![image](https://github.com/llamafilm/tesla-http-proxy-addon/assets/46457331/578bfc12-f799-4fd6-97b5-92bafecd5431)
Needed to create .pem files manually and put them in /ssl/

**Config file content**

`<VirtualHost *:8098>
ServerName teslaapi.<domain>

    Alias "/.well-known/appspecific/" "/share/tesla/"
    <Location "/.well-known/appspecific/">
        Options Indexes FollowSymLinks
        Allow from all
    </Location>
</VirtualHost>
`

**Comment by baylanger** [comment]
@SlowCar999 Check discussion #23 hoping it can help you.
